sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -1122,3 +1122,104 @@ func Test_connection_IsExternal(t *testing.T) {
 		})
 	}
 }
+
+func Test_getConnection_direction(t *testing.T) {
+	tests := []struct {
+		localPort  uint32
+		remotePort uint32
+		protocol   storage.L4Protocol
+		role       sensor.ClientServerRole
+		expected   bool
+	}{
+		{
+			localPort:  0,
+			remotePort: 53,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_UDP,
+			role:       sensor.ClientServerRole_ROLE_CLIENT,
+			expected:   false,
+		}, {
+			localPort:  0,
+			remotePort: 53,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_UDP,
+			role:       sensor.ClientServerRole_ROLE_SERVER,
+			expected:   false,
+		}, {
+			localPort:  53,
+			remotePort: 0,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_UDP,
+			role:       sensor.ClientServerRole_ROLE_CLIENT,
+			expected:   true,
+		}, {
+			localPort:  53,
+			remotePort: 0,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_UDP,
+			role:       sensor.ClientServerRole_ROLE_SERVER,
+			expected:   true,
+		}, {
+			localPort:  50000,
+			remotePort: 53,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_UDP,
+			role:       sensor.ClientServerRole_ROLE_CLIENT,
+			expected:   false,
+		}, {
+			localPort:  50000,
+			remotePort: 53,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_UDP,
+			role:       sensor.ClientServerRole_ROLE_SERVER,
+			expected:   false,
+		}, {
+			localPort:  53,
+			remotePort: 50000,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_UDP,
+			role:       sensor.ClientServerRole_ROLE_CLIENT,
+			expected:   true,
+		}, {
+			localPort:  53,
+			remotePort: 50000,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_UDP,
+			role:       sensor.ClientServerRole_ROLE_SERVER,
+			expected:   true,
+		}, {
+			localPort:  50000,
+			remotePort: 53,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_TCP,
+			role:       sensor.ClientServerRole_ROLE_CLIENT,
+			expected:   false,
+		}, {
+			localPort:  50000,
+			remotePort: 53,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_TCP,
+			role:       sensor.ClientServerRole_ROLE_SERVER,
+			expected:   true,
+		}, {
+			localPort:  53,
+			remotePort: 50000,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_TCP,
+			role:       sensor.ClientServerRole_ROLE_CLIENT,
+			expected:   false,
+		}, {
+			localPort:  53,
+			remotePort: 50000,
+			protocol:   storage.L4Protocol_L4_PROTOCOL_TCP,
+			role:       sensor.ClientServerRole_ROLE_SERVER,
+			expected:   true,
+		},
+	}
+
+	for _, tt := range tests {
+		networkConn := sensor.NetworkConnection{
+			SocketFamily: sensor.SocketFamily_SOCKET_FAMILY_IPV4,
+			LocalAddress: &sensor.NetworkAddress{
+				Port: tt.localPort,
+			},
+			RemoteAddress: &sensor.NetworkAddress{
+				Port: tt.remotePort,
+			},
+			Protocol: tt.protocol,
+			Role:     tt.role,
+		}
+		conn := getConnection(&networkConn)
+		assert.NotNil(t, conn)
+		assert.Equal(t, tt.expected, conn.incoming, "local: %d, remote: %d, protocol: %s, role: %s", tt.localPort, tt.remotePort, storage.L4Protocol_name[int32(tt.protocol)], sensor.ClientServerRole_name[int32(tt.role)])
```

Comment:
Maybe not for this test, but a new one: Wouldn't we also be interested in the correctness of the port, or is it only the directionality we're interested in?

Reply:
Well, what would be the definition of correctness? 

Strictly speaking, a port is just a number, any port can be listened on, there's no right or wrong, there's only some reserved ports on the lower ranges for specific applications and some guidelines OSes use for assigning random ports when an application doesn't specify it. Other than that, anything is fair game, port 8080 is usually used for an application to listen on, but nothing prevents you from using that same port for sending data if it's unused.

Reply:
I see, my question was too broad, let me try again :sweat_smile: 
We're checking for the correct value of `incoming`, but we don't look at the local or remote ports. 
Is that not important for this function, or any uses of the returned struct?  If not, my comment is a non-issue

Reply:
Ah, I see. Well the translation of the port is [simple enough](https://github.com/stackrox/stackrox/blob/73c7beec618fd5ebdaf6bbaa3e04e3833448380d/sensor/common/networkflow/manager/manager_impl.go#L1092), so I don't see much value in testing that. The ip addresses are a bit more involved, but those already have tests [here](https://github.com/stackrox/stackrox/blob/master/pkg/net/addr_test.go), so it feels like I'd be sort of duplicating those. WDYT? Is there a specific test case you have in mind?

Reply:
No, that's great as-is, thanks!